src/neo/SmartContract/ApplicationEngine.Storage.cs

```diff
@@ -93,17 +94,17 @@ private void PutExInternal(StorageContext context, byte[] key, byte[] value, Sto
             StorageItem item = Snapshot.Storages.GetAndChange(skey);
             if (item is null)
             {
+                newDataSize = key.Length + value.Length;
                 Snapshot.Storages.Add(skey, item = new StorageItem());
-                AddGas((key.Length + value.Length) * StoragePrice);
             }
             else
             {
                 if (item.IsConstant) throw new InvalidOperationException();
-                if (value.Length <= item.Value.Length)
-                    AddGas((1 + value.Length / 4) * StoragePrice);
-                else
-                    AddGas((1 + item.Value.Length / 4 + value.Length - item.Value.Length) * StoragePrice); // +1 means base fee
+                newDataSize = (1 + value.Length / 4); // +1 means base fee
+                if (value.Length > item.Value.Length)
+                    newDataSize += value.Length - item.Value.Length;
```

Comment:
Mistake? 
If `(value.Length > item.Value.Length` , then `newDataSize ` is `1 + value.Length / 4 + value.Length - item.Value.Length`, wihch is not equal to `1 + item.Value.Length / 4 + value.Length - item.Value.Length`

Reply:
Yes, Thanks! :)